Login page with MACC Post data and Sector data integrated

vennela-miryala opened the pull request.

shelagh-mcgowan committed:
Create README.md

shelagh-mcgowan committed:
Delete PeaceTrack Mobile App Requirements v2.docx

hopefully this just deletes from my fork...

shelagh-mcgowan committed:
Delete 4  Types of Activities Diagram.pdf

shelagh-mcgowan committed:
Delete Project Data Framework.pdf

shelagh-mcgowan committed:
Delete Activity Tracking Booklet_2014_3_10.docx

shelagh-mcgowan committed:
Delete Project Framework Example - Thailand Edu.docx

shelagh-mcgowan committed:
Delete Project Framework Example - Thailand Youth Dev.docx

shelagh-mcgowan committed:
Delete Project Framework Indicator List - Full - March 2014.xlsx

shelagh-mcgowan committed:
Delete Project Framework Indicator List - Thailand - April 2014.xlsx

shelagh-mcgowan committed:
Delete VRT 4 0_Proposal Outline_2014 03 24.docx

shelagh-mcgowan committed:
Delete VRT 4.0_Proposal_2014.03.24.pptx

shelagh-mcgowan committed:
Delete README.md

shelagh-mcgowan committed:
login functionality

shelagh-mcgowan committed:
will this version run?

shelagh-mcgowan committed:
troubleshooting

shelagh-mcgowan committed:
Added home view, 'cohort view', 'add cohort' view, as well as delete cohort and edit cohort name functionality.

shelagh-mcgowan committed:
attempt to fix issue

rosariorobinson committed:
Merge pull request #1 from shelagh-mcgowan/master

Create README.md

vennela-miryala committed:
Edited code to work with latest version of Xcode and Swift compiler. Structured the repo a bit too

vennela-miryala committed:
Edited code to work with latest version of Xcode and Swift compiler. Structured the repo a bit too

vennela-miryala committed:
Edited code to work with latest version of Xcode and Swift compiler. Structured the repo a bit too

vennela-miryala committed:
Edited code to work with latest version of Xcode and Swift compiler. Structured the repo a bit too

vennela-miryala committed:
Merge remote-tracking branch 'origin/master'

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
moved to main folder

vennela-miryala committed:
login and home screen ui done

vennela-miryala committed:
fixing directories

vennela-miryala committed:
fixing directories

vennela-miryala committed:
committing login code

vennela-miryala committed:
fixing directories

vennela-miryala committed:
adding the login code

vennela-miryala committed:
fixing directories

vennela-miryala committed:
adding the login code

kangaroo5383 commented:
As first priority:
@vennela-miryala please make sure to fix #8 

Some questions:
1. Why are there multiple `.xcodeproj` ?  There should only be one
2. The one in `GSoC-2015/PeaceTrack.xcodeproj` I'm assuming is the one you're working on? (good job on the correct structure!  The other ones seemed incorrect.)  However, I couldn't get it to compile, for pull requests, the expectation is that it should be compiling and working on download.

Let me know!

vennela-miryala commented:
@kangaroo5383 working on issue #8 

To answer your question:
1. The repo currently holds last years' project code just for reference. Once we are done with the complete implementation we'll remove that from the repo.
2. I cloned the repo on a different system and it is compiling for me. Please make sure you have the latest version of Xcode and Swift compiler. If you still face an issue, please let me know.

kangaroo5383 commented:
@vennela-miryala in your email you mentioned that you have XCode 6.2, the latest XCode is 6.4 and Swift compiler 1.2 can you update and confirm that it's working for you? 

Here are the errors I'm seeing

![image](https://cloud.githubusercontent.com/assets/34344/8451190/2881cb20-1f96-11e5-9d08-b535481c52eb.png)

vennela-miryala commented:
@kangaroo5383 I am updating my Xcode but the setup is of 2.61GB, and it might take a day to update as the internet connection here is slow.

From the screenshot it seems that all of them are soft compiler errors, so if you could click on each of those red dots, Xcode will automatically fix it and you should be able to build and test the implementation.

I'll fix it as soon as Xcode updates on my system.

kangaroo5383 commented:
Yes, I'm pretty sure I can resolve those compiler errors myself, it would be more beneficial if you are able to fix them.  Let me know when you have the latest Xcode and have updates that I can look it.

kangaroo5383 commented:
@vennela-miryala any update on this?

vennela-miryala committed:
finxed the errors

vennela-miryala committed:
fixing errors

vennela-miryala committed:
fixing the errors

vennela-miryala committed:
fixed errors

vennela-miryala commented:
@kangaroo5383 i have removed the compiler errors and added the new .gitignore file as well. Will refractor the getPosts code, currently working on Cohorts management workflow.

vennela-miryala committed:
cohorts implemented

kangaroo5383 commented:
@vennela-miryala do you have the more recent commits?  I'm looking at your latest from July here and there are still 26 compiler errors on Xcode 6.4

vennela-miryala commented:
@kangaroo5383 yea, this is not the latest code, will push it in a day or two. Me and the PeaceTrack Android student are yet to integrate the cohort management with MACC. Waiting for the MACC student to give us the proper CURL requests and I'll push the code right away. Till then we are syncing the UI and stuff among both the apps. I'll push the final code before our demo i.e. coming Wednesday for review

kangaroo5383 commented:
Sounds good @vennela-miryala I'll review it then, look forward to it.

ajay-sai committed:
Login, Cohorts, Activity, Measurment implemented

vennela-miryala committed:
clean up

vennela-miryala committed:
Activity, Cohort, Measurement implemented

kangaroo5383 commented:
There are a couple initial issues:
1. When the app starts - it shows this screen, however i've never signed in.  Shouldn't the 1st screen be log in screen?
![image](https://cloud.githubusercontent.com/assets/34344/9384588/2645f5d4-475a-11e5-8d9e-0efca620e55d.png)
1. When it's on this screen, the application is frozen waiting for `getPosts()` in `LoginViewController` to finish executing - mainly line 37 `sendSynchronousRequest`
   I would recommend either making it asynchronous so it doesn't block the UI or add a spinner.  ( see https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIActivityIndicatorView_Class/ )
2. For logging in the username and password looks to be hardcoded - is that expected? I would think that individuals were suppose to sign in with their own user name and password?
   `
       // set up the base64-encoded credentials
       let username = "admin"
       let password = "mypassword"
   `
3. General notes - any time I click on a button, the application froze, it would be good to trigger the activity indicator to show that it is doing something.  
4. In the accounts page, it's not possible to scroll down to enter the rest of the fields.  You should use scrollview so it's scrollable  
   ![image](https://cloud.githubusercontent.com/assets/34344/9384746/33cf04ec-475b-11e5-8911-7b35df157617.png)
5. The post selection with the `UIPickerView` is crashing with no content showing.
   ![image](https://cloud.githubusercontent.com/assets/34344/9384856/c2347c4e-475b-11e5-828b-04af042a7a8c.png)
6.  Are there test content that I can enter for that screen to see the loading of the rest of the content?

vennela-miryala commented:
1. I think it showed you directly the Home screen because you might have loaded the app in the same simulator before (previous build). If you reset the simulator once, it should work fine and you'll see the Login page load. What i'm basically doing is that it check on the Home page whether the "USERLOGGEDIN" key is set to 1(i.e. logged in) in the shared preferences of the app or not. If not then I take the user to the Login page automatically.
2. No
   // set up the base64-encoded credentials let username = "admin" let password = "mypassword"
   These credentials are used to make the POST request on the server. They are not a user's credentials. User enters the credentials in the text boxes and that data is sent to the server to validate.
3. Wasn't aware of such functionality. Will look into it.
4. Will add scrollview

kangaroo5383 commented:
1. no, I did not, also I just did a simulator reset, and it's still showing that screen as the 1st screen, then after about more than a minute it then goes to the login page.  This is related to the blocking call from `sendSynchronousRequest`
2. Great, Can you give me a sample user's login and password so I can verify the validation post requests
3. great! :) 
4. :+1:

vennela-miryala commented:
1. I just checked with the MACC student, it is blocking because they have turned down the IP I was sending requests to. They are pushing the code to production and will give the new IP.
2. Credentials given by MACC where:
   Name: Ajay Sai
   Email: sai.ajaysai@gmail.com
   Post: Thailand
   Sector: Education

It might not work now as the server is down. I have a demo prepared for today's meeting. Will change the IPs as soon as MACC updates them.

rosariorobinson commented:
Can we get this resolved? @ajay-sai @vennela-miryala please work together. Thanks.

vennela-miryala commented:
Ajay has sent out a mail to Julia to review and deploy his code on the main server. She do it by today most probably. Once that is done I'll change the IP address and push the code right away.

kangaroo5383 commented:
@vennela-miryala what's the status on the final readiness of this?

vennela-miryala commented:
@kangaroo5383, @ajay-sai said that Julia has deployed his code on the main server but Irish (from Malaria Prevention) and Ajay's code is interfering in someway and that is why its facing some issues. They are still trying to resolve it.
